src/onegov/org/app.py

```diff
@@ -193,23 +193,25 @@ def ticket_permissions(self):
         return result
 
     @orm_cached(policy='on-table-change:pages')
-    def homepage_pages(self) -> dict[int, list[Topic]]:
-
+    def topics_cached(self) -> list[tuple[int, 'Topic']]:
         def visit_topics(
-            pages: 'Iterable[Page]',
-            root_id: int | None = None
+            pages: 'Iterable[Page]', root_id: int | None = None
         ) -> 'Iterator[tuple[int, Topic]]':
             for page in pages:
+
                 if isinstance(page, Topic):
                     yield root_id or page.id, page
 
                 yield from visit_topics(
                     page.children,
-                    root_id=root_id or page.id
+                    root_id=root_id
                 )
 
+        return list(visit_topics(self.root_pages))
```

Comment:
```suggestion
        return tuple(visit_topics(self.root_pages))
```
`list` is mutable, so it can't be cached properly (it will work in the second-tier redis cache, but not the first tier request cache for subsequent calls in the same request), you need to use a hashable type, such as `tuple`.